src/pages/home/components/Sidebar/components/SocialAccordion/SocialAccordion.spec.tsx

```diff
@@ -8,82 +9,131 @@ import {
 
 vi.spyOn(window, 'scrollTo');
 
-// TODO: revisitar esse teste pois está falhando
-describe.skip('SocialAccordion', () => {
-  describe('when initilize', () => {
-    it('renders the component', () => {
+const mockDiscordData = mockedAccounts().data.DISCORD_EXAMPLE_ID;
+
+describe('SocialAccordion', () => {
+  it('renders the component', () => {
+    render(
+      <SocialAccordion
+        accounts={mockDiscordData}
+        error={false}
+        socialMediaId="DISCORD_EXAMPLE_ID"
+      />
+    );
+    const accordion = screen.getByText(/discord/i);
+
+    expect(accordion).toBeInTheDocument();
+  });
+
+  it('renders the intern content of accordion when is open', async () => {
+    render(
+      <SocialAccordion
+        accounts={mockDiscordData}
+        error={false}
+        socialMediaId="DISCORD_EXAMPLE_ID"
+      />
+    );
+
+    const accordion = screen.getByText(/discord/i);
+    await userEvent.click(accordion);
+
+    const innerContent = screen.getByText(mockDiscordData[0].userName);
+
+    expect(innerContent).toBeInTheDocument();
+  });
+
+  it('shows the error on screen if error={true}', () => {
+    render(
+      <SocialAccordion accounts={[]} error socialMediaId="DISCORD_EXAMPLE_ID" />
+    );
+    const error = screen.getByText(/error/i);
+
+    expect(error).toBeInTheDocument();
+  });
+
+  describe('social tab switch', () => {
+    it('activates social tab when is enable', async () => {
       render(
         <SocialAccordion
-          accounts={mockedAccounts().data.DISCORD_EXAMPLE_ID}
+          accounts={mockDiscordData}
           error={false}
           socialMediaId="DISCORD_EXAMPLE_ID"
         />
       );
+
       const accordion = screen.getByText(/discord/i);
+      await userEvent.click(accordion);
 
-      expect(accordion).toBeInTheDocument();
-    });
+      const switchesComponent = screen.getAllByRole('checkbox');
+      await userEvent.click(switchesComponent[0]);
 
-    it('renders the intern content of accordion when is open', () => {
+      expect(mockAddAccount).toHaveBeenCalledWith(mockDiscordData[0]);
+    });
+    it('deactivates social tab when is disable', async () => {
       render(
         <SocialAccordion
-          accounts={mockedAccounts().data.DISCORD_EXAMPLE_ID}
+          accounts={mockDiscordData}
           error={false}
           socialMediaId="DISCORD_EXAMPLE_ID"
         />
       );
-      const innerContent = screen.getByText(/discord/i);
 
-      expect(innerContent).toBeInTheDocument();
-    });
+      const accordion = screen.getByText(/discord/i);
+      await userEvent.click(accordion);
 
-    it('shows the error on screen if error={true}', () => {
-      render(
-        <SocialAccordion
-          accounts={[]}
-          error
-          socialMediaId="DISCORD_EXAMPLE_ID"
-        />
-      );
-      const error = screen.getByText(/error/i);
+      const switchesComponent = screen.getAllByRole('checkbox');
+      const [firstAccountSwitch] = switchesComponent;
```

Comment:
isso aqui nao funciona nao?

```suggestion
      const [firstAccountSwitch] = screen.getAllByRole('checkbox');
```